controllers/checlusterrestore/backup_data_restorer.go

```diff
@@ -316,11 +326,13 @@ func restoreCheCR(rctx *RestoreContext, dataDir string) (bool, error) {
 
 	if err := rctx.r.client.Create(context.TODO(), cheCR); err != nil {
 		if errors.IsAlreadyExists(err) {
-			return false, rctx.r.client.Delete(context.TODO(), cheCR)
+			return true, nil
```

Comment:
Add a few lines comment, please, that describes when such a situation possible and what this return mean.